### PR TITLE
Move argo fake client to Tekton fake client. Also update apiserver, common, and crd tests

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Related #388

**Description of your changes:**
Move argo fake client to Tekton fake client and update apiserver, common, and crd tests. Many of the removed tests are heavily typed with Argo. Pending the followings to completely remove argo client.
- agent (Evan working on it)
- cache (Peng is making changes, can temporary comment out all the argo section for now.)
- crd <del>(some test codes are not matching, will do a follow up PR to match with the unit tests)</del> (done)

Also added new Travis job to run the new unit tests without argo.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by Tomcli** [comment]
@ckadner the valid link test keeps failing because one of the twitter links is rate limited. Is there a way we could bypass that?

**Comment by ckadner** [comment]
> @ckadner the valid link test keeps failing because one of the twitter links is rate limited. Is there a way we could bypass that?

@Tomcli -- I created PR #396 to relax the Markdown link verification to 404 errors only. Ignoring rate-limited URLs.

File: .travis.yml
Comment (ckadner) [inline .travis.yml:68]: I think this should be called "Run go unit tests"? The tests are not _subject_ to verification, the tests are _doing_ the verification, no?

File: backend/src/common/util/scheduled_workflow_test.go
Comment (ckadner) [inline backend/src/common/util/scheduled_workflow_test.go:27]: should this comment move down to line 117?

**Review by ckadner — CHANGES_REQUESTED** [review]
Hi @Tomcli -- thanks for adding comments where large code blocks were removed to help with future merges of the upstream KFP code. There are a few large blocks that still miss a comment, could you add those? Use a common comment (same wording) that we can search for so we may be able to exclude those reported merge conflicts in the future

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:129]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:173]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:667]: add a comment why it was removed to help with future KFP merges? the one above might not show in the Git conflict resolution

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:761]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:836]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:871]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:906]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:956]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_test.go:2423]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/resource/resource_manager_util_test.go
Comment (ckadner) [inline backend/src/apiserver/resource/resource_manager_util_test.go:31]: move this comment to line 150?

File: backend/src/apiserver/server/pipeline_server_test.go
Comment (ckadner) [inline backend/src/apiserver/server/pipeline_server_test.go:86]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/server/pipeline_upload_server_test.go
Comment (ckadner) [inline backend/src/apiserver/server/pipeline_upload_server_test.go:461]: add a comment why it was removed to help with future KFP merges?

File: backend/src/apiserver/server/util_test.go
Comment (ckadner) [inline backend/src/apiserver/server/util_test.go:111]: add a comment why it was removed to help with future KFP merges? I know there is a comment above, but that might not show in Git merge conflict resolution scope.

File: backend/src/common/util/formatter_test.go
Comment (ckadner) [inline backend/src/common/util/formatter_test.go:328]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:153]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:226]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:256]: add a comment why it was removed to help with future KFP merges?

File: backend/src/common/util/workflow_test.go
Comment (ckadner) [inline backend/src/common/util/workflow_test.go:308]: add a comment why it was removed to help with future KFP merges?

File: backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go
Comment (ckadner) [inline backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go:70]: add a comment why it was removed to help with future KFP merges?

File: backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go
Comment (ckadner) [inline backend/src/crd/controller/scheduledworkflow/client/workflow_client_test.go:219]: add a comment why it was removed to help with future KFP merges?

File: backend/src/crd/controller/scheduledworkflow/util/scheduled_workflow_test.go
Comment (ckadner) [inline backend/src/crd/controller/scheduledworkflow/util/scheduled_workflow_test.go:680]: add a comment why it was removed to help with future KFP merges?

**Comment by ckadner** [comment]
If we use the same wording for all those comments, then we can search for those the next time we merge upstream KFP code and we may be able to exclude those reported merge conflicts automatically

**Comment by animeshsingh** [comment]
/hold

until we come back in Jan before merging

File: .travis.yml
Comment (Tomcli) [reply .travis.yml]: done

**Comment by evan-hataishi** [comment]
/lgtm

I'll can add in the agent unit test command to the makefile when this is merged.

**Review by ckadner — APPROVED** [review]
Thanks for adding the comments Tommy

/lgtm

**Comment by animeshsingh** [comment]
/unhold

@Tomcli  @ckadner do we have a good handle being able to relicate these changes as we pull in latest ?

**Comment by ckadner** [comment]
> @Tomcli @ckadner do we have a good handle being able to replicate these changes as we pull in latest ?

I did a try run of merging in the latest changes from KFP 1.3.0 into master it shows me 895 files changed with **43 rejected conflicts**.

If I attempt the same KFP merge on top of this PR I get **56 rejected conflicts**. So roughly 30% more work for KFP merges going forward.

**Pull request merged**
